demo/src/utils.js

```diff
@@ -28,36 +28,41 @@ export const getKeyringPair = async (ss58Format) => {
   return keyringPair;
 };
 
-export const sendExtrinsic = (extrinsic, api, keyringPair) =>
-  new Promise((resolve, reject) => {
-    const signAndSend = async () => {
-      const unsub = await extrinsic.signAndSend(keyringPair, (result) => {
-        const { status, events, dispatchError } = result;
-        console.log("status.type: ", status.type);
-
-        if (status?.isFinalized) {
-          unsub();
-          if (!_.isNil(dispatchError)) {
-            reject(dispatchError);
-          }
-
-          const event = _.find(events, ({ event: eventData }) =>
-            api.events.system.ExtrinsicSuccess.is(eventData),
-          );
-          if (event) {
-            resolve({
-              blockHash: status?.asFinalized?.toString(),
-              events,
-              extrinsicHash: extrinsic.hash,
-            });
-          } else {
-            reject(new Error("The event.ExtrinsicSuccess is not found"));
-          }
-        }
-      });
-    };
-    signAndSend();
-  });
+export const sendExtrinsic = (api, extrinsic, keyPair) =>
+	new Promise((resolve, reject) => {
+		const signAndSend = async () => {
+			try {
+				const unsub = await extrinsic.signAndSend(keyPair, (result) => {
+					const { status, events, dispatchError } = result;
+					console.log("status.type: ", status.type);
+
+					if (status?.isFinalized) {
+						unsub();
+
+						if (dispatchError) {
+							return reject(new Error(dispatchError));
```

Comment:
I added return here, so it will reject and return immediately. The below code doesn’t need to run.

Reply:
dispatchError was a string during my testing, so `new Error(message)` worked. You can double check its type. I think it’s just a string.

Reply:
You are right. I made a mistake. The dispatch error needs `dispatchError.toString()` to convert to string. This version is better as it can show the error name.
```
		if (dispatchError) {
							if (dispatchError.isModule) {
								console.log(`dispatchError.isModule`, dispatchError.isModule);
								const metaError = api.registry.findMetaError(dispatchError.asModule);
								const { name, section } = metaError;

								console.log(`metaError`, metaError);
								return reject(new Error(`${section}.${name}`));
							}

							return reject(new Error(dispatchError.toString()));
						}
```